examples/qml_features/rust/src/nested_qobjects.rs

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+// ANCHOR: book_macro_code
+#[cxx_qt::bridge(cxx_file_stem = "nested_qobjects")]
+mod ffi {
+    // ANCHOR: book_extern_block
+    unsafe extern "C++" {
+        #[cxx_name = "InnerObject"]
+        type InnerObjectPtr = super::qobject::InnerObject;
+    }
+    // ANCHOR_END: book_extern_block
+
+    #[cxx_qt::qobject]
+    #[derive(Default)]
+    pub struct InnerObject {
+        #[qproperty]
+        counter: i32,
+    }
+
+    #[cxx_qt::qobject]
+    pub struct OuterObject {
+        #[qproperty]
+        inner: *mut InnerObjectPtr,
```

Comment:
Renamed to `CxxInnerObject` for now, as this then matches the how CXX has a CxxVector for the C++ side.

We also might want to consider renaming our internal `TQt` to something else as this would collide with custom definitions :thinking: